### DISCUSSION TITLE
Picker control

### DISCUSSION THREAD

**Discussion by kj4bix** [dotnet/maui#22724] [General] [answered]
I have a picker that can potentially have a large number of items in the list. I don't want to build the list until they request it. Instead of it being built initially or based on dependencies on other controls. Basically, build the picker list on demand. I can't seem to find an available event for when the user requests the list. Has anyone done this in the past?

**Comment by gautambjain** [comment]
Can you bind the picker's itemsource to a list? As shown here: https://stackoverflow.com/questions/74212915/how-do-you-bind-maui-picker-to-a-viewmodel

**Reply by kj4bix** [reply]
I want to build the list when the user clicks on the control, not preload the control. I have a list that takes a few seconds to get the data. I can't seem to find an event for when the user clicks on the picker. I bind other pickers, but the data is basically preloaded and not only loaded when the user clicks on the list. Maybe the only way is a custom control with a tap gesture. It would seem the control would have an event for this scenario.

**Comment by davidortinau** [comment, marked as answer]
How about using an empty `ObservableCollection` bound to the `Picker` and adding items to it in the `Focused` event? Attached sample:

[PickerFocused.zip](https://github.com/user-attachments/files/15554427/PickerFocused.zip)

**Reply by kj4bix** [reply]
Good idea, far better than a custom control. I'll take it out for a spin. Thanks David!

**Reply by kj4bix** [reply]
David, I did the Focused Event and it worked, except for it continuously getting hit. Yes, I could work around that by having the previous values that built the list and if they change rebuilt the list. But I think the easier approach was to add a TapGestureRecognizer in my XAML and invoke my method in the viewmodel. I get hit when they tap on the picker, seems to work really well. Just thought I would comment, thanks for getting me going that direction. 

Alan
